projects/style_gen/modules.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Models and helper classes for style-controlled generation.
+"""
+
+import random
+
+from parlai.core.torch_agent import History
+
+
+STYLE_SEP_TOKEN = ' STYLE '
+
+
+class StyleHistoryMixin(History):
```

Comment:
could we potentially use the TorchAgent funciton `get_temp_history` to do this instead? it was added after you wrote this code initially, but makes the changes much simpler

Reply:
Yeah I would love that

Reply:
Yes, good idea! So, I refactored `StyleAgentMixin` to use `get_temp_history()`, and I re-ran the command that I listed in `model_list.py`: `parlai eval_model --datatype test --model projects.style_gen.style_gen:StyleGenAgent --model-file zoo:style_gen/c75_labeled_dialogue_generator/model --skip-generation True --task style_gen:LabeledBlendedSkillTalk --use-style-frac 1.00`. The resulting ppl is 0.026 higher than before, and in fact I see that there are slight differences in some of the generations:

<img width="1893" alt="Screen Shot 2020-08-03 at 5 10 29 PM" src="https://user-images.githubusercontent.com/5393263/89228155-0c430a00-d5ad-11ea-9730-d10e9587fa45.png">

However, clearly the agent is still setting the style correctly, because if it weren't, the perplexities would be *way* higher than what I'm seeing (around 10.15, instead of 9.47). Thus, I'll chalk this up to fluctuations in generation